cryptographic_estimators/MAYOEstimator/mayo_estimator.py

```diff
@@ -66,68 +67,94 @@ def table(self, show_quantum_complexity=0, show_tilde_o_time=0,
-            +--------------+-------------+-------+--------+------------+
-            | algorithm    | attack_type |  time | memory | parameters |
-            +--------------+-------------+-------+--------+------------+
-            | DirectAttack |   forgery   | 274.7 |  175.0 | {'K': 24}  |
-            +--------------+-------------+-------+--------+------------+
-
-            sage: E = MAYOEstimator(n=90, m=56, o=8, k=10, q=16)
+            +----------------------+--------------+----------------------------------------------------+
+            |                      |              |                      estimate                      |
+            +----------------------+--------------+-------+--------+-----------------------------------+
+            | algorithm            | attack_type  |  time | memory |             parameters            |
+            +----------------------+--------------+-------+--------+-----------------------------------+
+            | DirectAttack         |   forgery    | 274.7 |  175.0 |             {'K': 24}             |
+            | KipnisShamir         | key-recovery | 460.9 |   19.7 |                 {}                |
+            | ReconciliationAttack | key-recovery | 275.7 |   95.3 | {'k': 18, 'variant': 'las_vegas'} |
+            | IntersectionAttack   | key-recovery | 525.2 |   59.6 |  {'k': 0, 'variant': 'las_vegas'} |
+            | ClawFinding          |   forgery    | 270.6 |  260.6 |    {'X': 258.415, 'Y': 253.585}   |
+            +----------------------+--------------+-------+--------+-----------------------------------+
+
+            sage: E = MAYOEstimator(n=90, m=56, o=8, k=10, q=16) # long test
             sage: E.table(show_all_parameters=1) # long time
-            +--------------+-------------+----------------------------+
-            |              |             |          estimate          |
-            +--------------+-------------+------+--------+------------+
-            | algorithm    | attack_type | time | memory | parameters |
-            +--------------+-------------+------+--------+------------+
-            | DirectAttack |   forgery   | 21.4 |   56.9 | {'K': 10}  |
-            +--------------+-------------+------+--------+------------+
-
-            sage: E = MAYOEstimator(n=64, m=60, o=10, k=21, q=16)
+            +----------------------+--------------+----------------------------------------------------+
+            |                      |              |                      estimate                      |
+            +----------------------+--------------+-------+--------+-----------------------------------+
+            | algorithm            | attack_type  |  time | memory |             parameters            |
+            +----------------------+--------------+-------+--------+-----------------------------------+
+            | DirectAttack         |   forgery    |  21.4 |   56.9 |             {'K': 10}             |
```

Comment:
This looks wrong, the numbers are far to good, and memory > time

Reply:
Nice catch!

@fmerino21 I checked the problem seems to be that. 
In the for the memory complexity we shouldn't add the factor `q^K`, where `K` is the number of variables to specialize.  Here we should assume that to solve each subsystem with `n-K` variables  the same  memory space  is used.

For instance, see how is handled in `hybrid_f5.py`